packages/cozy-stack-client/src/getIllegalCharacter.spec.js

```diff
@@ -0,0 +1,24 @@
+import { getIllegalCharacters } from './getIllegalCharacter'
+
+describe('getIllegalCharacters', () => {
+  it('should throw error if file name empty', () => {
+    expect.assertions(1)
+    const name = undefined
+    try {
+      getIllegalCharacters({ name })
+    } catch (error) {
+      expect(error.message).toEqual('file should have a name property ')
+    }
+  })
+
+  it('should return empty string when no illegal characters found', () => {
+    const name =
+      'Valid file name with #@&"\'()§è!çà)-°_^¨*$€ù%`£=+:;.,?∞…÷≠≤<>•ë“‘{¶«Çø}æê®†Úºîœπ‡Ò∂ƒﬁÌÏÈ¬µ‹≈©◊ß~ '
+    expect(getIllegalCharacters({ name })).toEqual('')
+  })
+
+  it('should return illegal characters split with spaces when illegal characters found', () => {
```

Comment:
nit: can we have a test method for each individual invalid char (in addition to this one)?

This would ensure that the regex does not ignore one of them (if we change it in the future).

Current test would pass even if the regex tests only for `\n`. But it tests results' concatenation, which is great, so do not remove it.